### DISCUSSION TITLE
Totals Per Device Per Day in kWh

### DISCUSSION THREAD

**Discussion by regsw** [jertel/vuegraf#28] [Ideas]
Hi, 
The real-time graphing is working really well but I would love to be able to track total usage in kWh over time. I have tried various queries to accomplish this but I can't get it to work properly. 

I think it would be great to be able to pull a total usage per circuit per day so that we can monitor total kWh over time. Then I could take that and sum it to weekly, monthly either by circuit or for whole house usage. 

I know that I can pull a CSV report from the Emporia app but I would think that this should be available via Grafana too but I can't seem to arrive on a query that works so I thought maybe there was something else that could be pulled from Emporia that would provide totals properly.

Thank you

**Comment by jertel** [comment]
What's stored in the Vuegraf DB is essentially WattSeconds, since it's recording the Watts used across each progressing second. If you want KwH for a selected Grafana date range, wouldn't you be able to take the sum of the WattSeconds and divide by 3,600,000 (3600 seconds in an hour x 1000 watts in a Kw)?

![image](https://user-images.githubusercontent.com/16639611/109198211-f32dc080-776b-11eb-9b42-adf8608b04b1.png)

I figure there is likely some double dipping going on, either in the data coming from Emporia, or perhaps Vuegraf is slightly overlapping (re-requesting) some intervals, since my numbers are always higher than my Electric bill for the same period.

I haven't focused much on accuracy of the data since my primary reason for building Vuegraf is for detecting relative usage trends and proactive alerting to abnormal energy usage. But if someone wants to double check my logic it wouldn't hurt.

**Reply by regsw** [reply]
Thanks for the info about the time. I ended up using that and having to multiply it by .503 to get it within an acceptable range of what Emporia shows on their app/CSV download. The other key for me was setting the timezone properly since I think some of the data was bleeding into other days causing additional errors.


![image](https://user-images.githubusercontent.com/15692711/109220676-3ea19800-7787-11eb-986f-56ff4722da86.png)

**Comment by EJansenMSFT** [comment]
I was also playing with this (but for tracking **_Avg, Min, Max_** kW usage during a given timeframe - so not exactly the same), but I had to change the data around to make it work.  For example, if I wanted the average power usage for an hour, I had to set the time range to "Last 1 Hour", ensure that the field defined for display had Watt(W), and that my math matched up with the timeframe:

![image](https://user-images.githubusercontent.com/72317762/122859835-a5286400-d2ea-11eb-8ac2-ebc04eb577e6.png)

Unfortunately, even that doesn't quite match up exactly with what I see on the Emporia App, but it's pretty close.

I would think that there's has to be some way that this math can automatically done.  For example, I noticed that in the 'Group By' field the time option has $interval defined in it, so if the $interval is 1 hour, then I figure there would be some way to change the value inside of math(/**_$interval.seconds_**), so that it would equal the interval in seconds - not sure what programming language or syntax is used; I just assumed that I could do _object.property_ .  Any idea how that could be done?  I don't know anything about Grafana or SQL databases (if Influx is even a SQL type database) at all.

Thanks,
Eric

**Reply by EJansenMSFT** [reply]
Actually, in playing around with it a bit, it appears that there's no need to do any math - options exist to do it for you:

![image](https://user-images.githubusercontent.com/72317762/122877996-3c021a00-d305-11eb-9b27-7d0fc13c9d58.png)


![image](https://user-images.githubusercontent.com/72317762/122878098-56d48e80-d305-11eb-876c-c61850dcd731.png)


Which is slightly different from the query used for the 'All Circuits' or 'Active Grid Usage' query in that you just omit the 'Group By' options:

![image](https://user-images.githubusercontent.com/72317762/122878267-8f746800-d305-11eb-847b-a4c084d06c50.png)

![image](https://user-images.githubusercontent.com/72317762/122878370-b03cbd80-d305-11eb-8a1c-b88c33d7a3d4.png)

HTH someone.

Cheers,
Eric

**Comment by EJansenMSFT** [comment]
Update...since the previous posts weren't exactly relevant to the title of the post - to get Kilowatt Hours for a timespan, (instead of tracking **_Avg/Min/Max_** loads like I was interested in / what I posted above), I did the following:

![image](https://user-images.githubusercontent.com/72317762/122992139-cd0fda00-d373-11eb-9fcd-990b9aa57030.png)

I changed the field to **Watt-Hour(Wh)** and then used Sum to add everything up for the applied $timeFilter and then divided by 3600 seconds and it matches up with what I see in the Emporia App.
